interp/constrexpr.mli

```diff
@@ -104,8 +104,9 @@ and cases_pattern_expr = cases_pattern_expr_r CAst.t
 and kinded_cases_pattern_expr = cases_pattern_expr * Glob_term.binding_kind
 
 and cases_pattern_notation_substitution =
-    cases_pattern_expr list *     (* for constr subterms *)
-    cases_pattern_expr list list  (* for recursive notations *)
+    cases_pattern_expr list *      (* for cases_pattern subterms parsed as terms *)
+    cases_pattern_expr list list * (* for recursive notations parsed as terms*)
+    kinded_cases_pattern_expr list (* for cases_pattern subterms parsed as binders *)
```

Comment:
Maybe this would be worth changing that to a record (but let's save that for a further PR).

Reply:
Actually, to support #7959, I have a next step which replaces this tuple of 4 different types by a map with a larger set of types. So, better indeed to merge this one first.